### PR TITLE
Support for treatment strategies with interventions at non-consecutiv…

### PR DESCRIPTION
**Pull request by jmafoster1**
Support for treatment strategies with interventions at non-consecutive timesteps. Timesteps without an explicitly specified intervention are treated as "don't care", so anything can happen.

### CODE REVIEW

**Comment by jmafoster1** [comment]
@f-allian, is the codecov just failing on this because the codebase has got slightly bigger? It says I have 100% coverage of the patch, so that's all I can think of, but adding in extra code with 100% coverage should be fine right?

File: causal_testing/specification/capabilities.py
[the remark this answers is not in the document]
Comment (jmafoster1) [reply causal_testing/specification/capabilities.py:1]: Yes, this was intentional. It is no longer required as the new functionality requires timesteps to be specified explicitly, which is what I wrote `capabilities.py` to handle implicitly.

**Review by f-allian — APPROVED** [review]
@jmafoster1 Feel free to merge following our meeting today :)

**Comment by jmafoster1** [comment]
Pretty sure I've fixed all the bugs in this. @f-allian , I've re-requested a review as I've changed quite a lot of stuff. Most of it is quite technical, but please let me know if anything seems odd.

**Review by f-allian — APPROVED** [review]
@jmafoster1 The new changes all look good to me

**Pull request merged**